Correct mixing of plain lists and headlines

hstock opened the pull request:
Retrieve the starting depth of a plain list as the level of the parent
headline. This way, plain lists and headlines can be mixed even if
lower level headlines get exportet as bullet points. (H: option)

yashi commented on ox-textile.el:
```diff
@@ -146,9 +146,13 @@ CONTENTS is the contents of the list.  INFO is a plist holding
+	   (org-export-get-parent-headline item) info))
+	 (headline-levels (plist-get info :headline-levels))
+	 (parent item)
+	 (depth (max 0 (- headline-level headline-levels))))
```
You can get the depth using `org-export-low-level-p`.  How about something like this:

```elisp
(let* ((parent item)
         (headline (org-export-get-parent-headline item))
	 (depth (or (and headline
                         (org-export-low-level-p headline info))
                    0)))
```
Is this feasible?

yashi commented on ox-textile.el:
```diff
@@ -147,9 +147,12 @@ contextual information."
-	   (org-export-get-parent-headline item) info))
+  (let* ((headline (org-export-get-parent-headline item))
+	 (headline-level
+	  (if headline
```
Since you are handling "no headline" case here, would you mind to add another test for that?

hstock replied:
I just checked - the "ordered-list" and "unordered-list" tests already fail without this fix since they test cases where no headline is included.

hstock commented:
I fixed the headline-mixed test so it really tests the change (it accidentally also worked with the old behaviour). 
Also added your proposed change using org-export-low-level-p. Thanks for the hint.
The "no headline" case is already covered by other tests, I just did not run the tests during first development stages, so I did not see that.

yashi commented:
This is super!  Thank you!
